modP2pImpl/test/org/aion/p2p/impl/comm/NodeMgrTest.java

```diff
@@ -292,89 +336,400 @@ public void test_getActiveNodesList() {
+                outbound.add(node);
+            }
+        });
+
+        Thread tMoveTempNodeToInbound = new Thread(() -> {
+            while (start.get()) {
+                try {
+                    Thread.sleep(15);
+                } catch (InterruptedException e) {
+                    e.printStackTrace();
+                }
+                INode node = nMgr.tempNodesTake();
+                if (node == null) {
+                    continue;
+                }
+                nMgr.addInboundNode(node);
+                //noinspection unchecked
+                inbound.add(node);
+            }
+        });
+
+        Thread tMovePeerToActive = new Thread(() -> {
+            while (start.get()) {
+                int i = r.nextInt(2);
+                if (i == 0) {
+                    INode node = null;
+                    try {
+                        node = (INode) inbound.take();
+                    } catch (InterruptedException e) {
+                        e.printStackTrace();
+                    }
+                    assert node != null;
+                    nMgr.movePeerToActive(node.getChannel().hashCode(), "inbound");
+                } else {
+                    INode node = null;
+                    try {
+                        node = (INode) outbound.take();
+                    } catch (InterruptedException e) {
+                        e.printStackTrace();
+                    }
+                    assert node != null;
+                    nMgr.movePeerToActive(node.getIdHash(), "outbound");
+                }
+
+                count.getAndDecrement();
+
+                try {
+                    Thread.sleep(8);
+                } catch (InterruptedException e) {
+                    e.printStackTrace();
+                }
+            }
+        });
+
+        Thread tDropActive = new Thread(() -> {
+            while (start.get()) {
+
+                HashMap activeMap = nMgr.getActiveNodesMap();
+
+                Iterator it = activeMap.entrySet().iterator();
+                if (it.hasNext()) {
+                    Entry en = (Entry) it.next();
+                    nMgr.dropActive((Integer) en.getKey(), "test");
+                }
+
+                count.getAndDecrement();
+
+                try {
+                    Thread.sleep(100);
+                } catch (InterruptedException e) {
+                    e.printStackTrace();
+                }
+            }
+        });
+
+        start.set(true);
+
+        tGenTempNode.start();
+
+        Thread.sleep(100);
+        tMoveTempNodeToOutbound.start();
+        tMoveTempNodeToInbound.start();
+        tMovePeerToActive.start();
+        tDropActive.start();
+
+        while (count.get() > 0) {
+            System.out.println(
+                "Node counts: " + count.get() + " activeSize: " + nMgr.getActiveNodesList().size()
+                    + " tempSize: " + nMgr.tempNodesSize());
+            Thread.sleep(1000);
```

Comment:
Is there a meaningful assert that can be added here? Perhaps consider using the `org.aion.db.impl.DatabaseTestUtils.assertConcurrent` implementation.

Reply:
okay, will refactor it to use assertConcurrent